libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -282,10 +282,14 @@ export class Subspace {
   public getPnpmfilePath(): string {
     const subspaceConfigFolderPath: string = this.getSubspaceConfigFolder();
 
-    return path.join(
-      subspaceConfigFolderPath,
-      (this._rushConfiguration.packageManagerWrapper as PnpmPackageManager).pnpmfileFilename
-    );
+    let pnpmFilename: string = (this._rushConfiguration.packageManagerWrapper as PnpmPackageManager)
```

Comment:
It's not clear that `packageManagerWrapper` can be safely casted to `PnpmPackageManager` here.

However, it looks like this cast was originally introduced by @D4N14L in fa58365c1af8afa55c057b263e6c3f7d724a306b, so out of scope for this PR.

Daniel's diff includes some other casts that do consider the package manager type:

```ts
      if (packageManagerWrapper.packageManager === 'pnpm') {
        knownSet.add((packageManagerWrapper as PnpmPackageManager).pnpmfileFilename.toUpperCase());
      }
```

Perhaps he overlooked this requirement from the API docs for `getPnpmfilePath()`:

> The file path is returned even if PNPM is not configured as the package manager.

Now that the path depends on the version of PNPM, this requirement seems impossible.  So maybe we should change the API to throw an exception if invoked for a different package manager. I bet we could establish that it is unlikely to cause any significant regressions.

@iclanton